flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartition.java

```diff
@@ -258,24 +225,13 @@ public void flush(int subpartitionIndex) {
 	 */
 	@Override
 	public void finish() throws IOException {
-		boolean success = false;
-
-		try {
-			checkInProduceState();
+		checkInProduceState();
 
-			for (ResultSubpartition subpartition : subpartitions) {
-				subpartition.finish();
-			}
-
-			success = true;
+		for (ResultSubpartition subpartition : subpartitions) {
+			subpartition.finish();
 		}
-		finally {
```

Comment:
Could you double check if there are no `RuntimeException`s that we are expecting to be thrown from `subpartition.finish()`? I think that would be the only reason for having this strange `try/finally/success` thingy.

Reply:
Yes, I also checked the related processes and it seems no explicit `RuntimeException` here.